File: javascript/src/models/rabbitMqConfigIn.ts
// this file is @generated

/** Configuration for a RabbitMq sink. */
export interface RabbitMqConfigIn {
  uri: string;
  routingKey: string;
}

export const RabbitMqConfigInSerializer = {
  _fromJsonObject(object: any): RabbitMqConfigIn {
    return {
      uri: object["uri"],
      routingKey: object["routingKey"],
    };
  },

  _toJsonObject(self: RabbitMqConfigIn): any {
    return {
      uri: self.uri,
      routingKey: self.routingKey,
    };
  },
};

File: rust/src/models/rabbit_mq_config_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

/// Configuration for a RabbitMq sink.
#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct RabbitMqConfigIn {
    pub uri: String,

    #[serde(rename = "routingKey")]
    pub routing_key: String,
}

impl RabbitMqConfigIn {
    pub fn new(uri: String, routing_key: String) -> Self {
        Self { uri, routing_key }
    }
}
